File: src/proficiency/math/herons_formula/herons.clj
(ns proficiency.math.herons_formula
  (:require [clojure.math.numeric-tower :as math]))

(defn semiperimeter [a b c]
  (/ (+ a b c) 2))

(defn area-of-triangle [a b c]
  (def s (semiperimeter a b c))
  (math/sqrt (* s (- s a) (- s b) (- s c))))

(let [[a b c] (map #(Integer. %) *command-line-args*)]
  (println (area-of-triangle a b c)))

File: src/proficiency/math/herons_formula/Herons.java
import java.lang.Math;

public class Herons {
    public static double semiperimeter(int a, int b, int c) {
        return (double) (a + b + c)/2;
    }

    public static double areaOfTriangle(int a, int b, int c) {
        double s = semiperimeter(a, b, c);
        return Math.sqrt(s * (s-a) * (s-b) * (s-c));
    }

    public static void main(String[] args) {
        int a = Integer.parseInt(args[0]);
        int b = Integer.parseInt(args[1]);
        int c = Integer.parseInt(args[2]);

        double area = areaOfTriangle(a, b, c);
        System.out.println(area);
    }
}
